### ISSUE TITLE
sssh-ssshd.scm test fails with the default crypto policy of Fedora 33

### ISSUE THREAD

**Issue by lantw44** [artyom-poptsov/guile-ssh#26]
Console output:
```
make[3]: Entering directory '/builddir/build/BUILD/guile-ssh-0.13.1/tests'
PASS: log.scm
PASS: server.scm
PASS: session.scm
PASS: key.scm
FAIL: sssh-ssshd.scm
PASS: shell.scm
PASS: tunnel.scm
PASS: popen.scm
PASS: dist.scm
PASS: server-client.scm
PASS: client-server.scm
============================================================================
Testsuite summary for Guile-SSH 0.13.1
============================================================================
# TOTAL: 11
# PASS:  10
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See tests/test-suite.log
Please report to poptsov.artyom@gmail.com
============================================================================
make[3]: *** [Makefile:676: test-suite.log] Error 1
```
File tests/sssh-ssshd.log:
```
%%%% Starting test sssh-ssshd
Group begin: sssh-ssshd
Test begin:
  test-name: "ssshd, start"
  source-file: "tests/sssh-ssshd.scm"
  source-line: 97
  source-form: (test-assert "ssshd, start" (let ((max-tries 10)) (system *ssshd-cmd*) (let ((pid (wait-pid-file max-tries *srv-pid-file*))) (cleanup pid) pid)))
Test end:
  result-kind: pass
  actual-value: 4356
Test begin:
  test-name: "sssh, exec"
  source-file: "tests/sssh-ssshd.scm"
  source-line: 104
  source-form: (test-assert "sssh, exec" (let ((max-tries 10)) (system *ssshd-cmd*) (let* ((pid (wait-pid-file max-tries *srv-pid-file*)) (output (read-line (open-input-pipe *test-cmd*))) (p (open-input-pipe *sssh-cmd*)) (result (let r ((res "") (l (read-line p))) (and (not (eof-object? l)) (if (string=? output res) #t (r (string-append res l) (read-line p))))))) (cleanup pid) result)))
Test end:
  result-kind: fail
  actual-value: #f
Group end: sssh-ssshd
# of expected passes      1
# of unexpected failures  1
e:              ssshd.pid
%%%% Starting test sssh-ssshd  (Writing full log to "sssh-ssshd.log")
tests/sssh-ssshd.scm:104: FAIL sssh, exec
# of expected passes      1
# of unexpected failures  1
FAIL sssh-ssshd.scm (exit status: 1)
```
Since the log file provided no useful information, I tried to debug it by modifying the build script. It seems that the failed test expected `read-line` to return the output of `uname` command, but most of the time it got `The key algorithm 'ssh-rsa' is not allowed to be used by PUBLICKEY_ACCEPTED_TYPES configuration option` instead. It suggests it is related to the crypto policy. If I ran `update-crypto-policies --set LEGACY` before running `make check`, the test passed.

Fedora 33 crypto policy changes: https://fedoraproject.org/wiki/Changes/StrongCryptoSettings2.

**Comment by artyom-poptsov** [comment]
Hello,

please check Guile-SSH from the `master` branch -- the problem should be fixed by now in 79e5185

Artyom.

**Comment by lantw44** [comment]
I tested it by applying commit 79e51857d4ce98dc13b723c874b6793cb68a4254 as a patch on Guile-SSH 0.13.1. All tests passed on Fedora 34.
